critiquebrainz/frontend/forms/review.py

```diff
@@ -43,12 +43,22 @@ class ReviewEditForm(Form):
         ],
         validators=[validators.DataRequired(message=lazy_gettext("You need to choose a license!"))])
     language = SelectField(lazy_gettext("You need to accept the license agreement!"), choices=languages)
+    rating = IntegerField(lazy_gettext("Rating"), widget=Input(input_type='number'), validators=[validators.Optional()])
 
     def __init__(self, default_license_id='CC BY-SA 3.0', default_language='en', **kwargs):
         kwargs.setdefault('license_choice', default_license_id)
         kwargs.setdefault('language', default_language)
         Form.__init__(self, **kwargs)
 
+    def validate(self):
+        if not super(ReviewEditForm, self).validate():
+            return False
+        if not self.text.data and not self.rating.data:
+            message = "Please provide at least one of text or rating for this review."
```

Comment:
The message isn't entirely grammatical. I suggest "Please provide some text or a rating for this review."